hibernate-core/src/main/java/org/hibernate/query/spi/QueryEngine.java

```diff
@@ -127,6 +131,11 @@ public QueryEngine(
 			userDefinedRegistry.overlay( sqmFunctionRegistry );
 		}
 
+		for ( FunctionContributor contributor : serviceRegistry.getService( ClassLoaderService.class )
```

Comment:
I think we should introduce a notion of ordering for such contributors to avoid non-determinism. IMO the `FunctionContributor` interface should have a default method:

```
default int ordinal() {
    return 1000;
}
```

With a comment that system provided contributors have an ordinal from e.g. 0 - 500, library provided ones should have an ordinal from 500 - 1000 and user provided ones should have an ordinal of 1000+.

The loaded services should then be put into a list, and sorted by a comparator that sorts first by ordinal and second by fully qualified class name.